### ISSUE TITLE
ecosystem: Do we want to build a S3 package for Synology?

### ISSUE THREAD

**Issue by nl5887** [minio/minio#4210]
Synology is a very popular NAS storage for (small) business and home use. If we create a package (https://www.synology.com/en-us/support/developer) to install Minio on this appliance we'll give synology users instantly s3 compatible object storage.

**Comment by mvnixon** [comment]
Would be great!

**Comment by karlbunch** [comment]
+1

**Comment by jahands** [comment]
I'd love this!

**Comment by DanLauer** [comment]
+1

**Comment by cynthia** [comment]
It is worth noting that the cross compilation capabilities Go provides is nearly not enough to offer full coverage across all Synology product lines. It is possible to install via ssh and autostart with standard init trickery - I've just tried the official ARM32 build (and a freshly build ARM32 build) on my 416slim and it seems to work just fine, FWIW.

**Comment by agidi** [comment]
Love the idea of an officially deployed package a+++

**Comment by ebartz** [comment]
+1

**Comment by cledorze** [comment]
+1, would bring easy S3 access to all homelabs owners who already got a Syno .

**Comment by BrianGilbert** [comment]
+1

**Comment by kannappanr** [comment]
We do not have access to a box. If someone from the community is willing to implement it, we welcome it.
Closing this issue for now.

**Comment by haydonryan** [comment]
+1

**Comment by ebartz** [comment]
There is actually a Solution to this. All the Synology Plus boxes which are based on a x86 CPU support running Docker. So to get this running you would have to launch a Docker container on your Synology and you are good to go.

**Comment by BrianGilbert** [comment]
Yes Docker is the correct way to do this, but you do have to start it from terminal instead of the docker GUI..

Execution command:
```
    docker run -d -p 9000:9000 --name minio1 \
        -e "MINIO_ACCESS_KEY=YOURACCESSKEY" \
        -e "MINIO_SECRET_KEY=YOURSECRETKEY" \
        -v /volume1/docker/minio/data:/data \
        -v /volume1/docker/minio/config:/root/.minio \
        minio/minio server /data
```
Then create a scheduled task (assuming you have let's encrypt certs configured as the default for your Synology), with the following script so that the site will be accesible via https (restart the instance from Docker gui to enable the certs):
```
    cd /usr/syno/etc/certificate/_archive/ && cp -f /usr/syno/etc/certificate/_archive/$(cat DEFAULT)/privkey.pem /volume1/docker/minio/config/certs/private.key && cp -f /usr/syno/etc/certificate/_archive/$(cat DEFAULT)/fullchain.pem /volume1/docker/minio/config/certs/public.crt

**Comment by zh99998** [comment]
+1

not all x86 synology device runs Docker.
my DS418play is x86, but don't have docker.

**Comment by BrianGilbert** [comment]
**Note:** I've just updated the scheduled task command above.

**Comment by darkmuggle** [comment]
I took a stab at packaging for Synology SPK's over https://github.com/darkmuggle/miniospk. Its really not useful beyond the scope of a trusted/home network (no SSL, none of the advanced features of Minio, etc). It works for my use-case. 

I debated whether to target the packaging for a PR to Minio, but ultimately felt that its packaging work. Synology SPK's is analogous to Debian or RPM packaging, and those recipes are not included in with Minio's source.
